pkg/canary/deployment_controller.go

```diff
@@ -197,6 +197,21 @@ func (c *DeploymentController) ScaleFromZero(cd *flaggerv1.Canary) error {
 		if primary.Spec.Replicas != nil && *primary.Spec.Replicas > 0 {
 			replicas = primary.Spec.Replicas
 		}
+	} else if cd.Spec.AutoscalerRef != nil {
+		hpa, err := c.kubeClient.AutoscalingV2().HorizontalPodAutoscalers(cd.Namespace).Get(context.TODO(), cd.Spec.AutoscalerRef.Name, metav1.GetOptions{})
+		if err == nil {
+			if hpa.Spec.MinReplicas != nil && *hpa.Spec.MinReplicas > 1 {
+				replicas = hpa.Spec.MinReplicas
+			}
+		} else {
+			// fallback to v2beta2
+			hpa, err := c.kubeClient.AutoscalingV2beta2().HorizontalPodAutoscalers(cd.Namespace).Get(context.TODO(), cd.Spec.AutoscalerRef.Name, metav1.GetOptions{})
+			if err == nil {
+				if hpa.Spec.MinReplicas != nil && *hpa.Spec.MinReplicas > 1 {
+					replicas = hpa.Spec.MinReplicas
+				}
+			}
+		}
```

Comment:
```suggestion
	if cd.Spec.AutoscalerRef.Kind == "HorizontalPodAutoscaler" {
        hpa, err := c.kubeClient.AutoscalingV2().HorizontalPodAutoscalers(cd.Namespace).Get(context.TODO(), cd.Spec.AutoscalerRef.Name, metav1.GetOptions{})
        if err == nil {
            if hpa.Spec.MinReplicas != nil && *hpa.Spec.MinReplicas > 1 {
                replicas = hpa.Spec.MinReplicas
            }
        } else {
            // fallback to v2beta2
            hpa, err := c.kubeClient.AutoscalingV2beta2().HorizontalPodAutoscalers(cd.Namespace).Get(context.TODO(), cd.Spec.AutoscalerRef.Name, metav1.GetOptions{})
            if err == nil {
                if hpa.Spec.MinReplicas != nil && *hpa.Spec.MinReplicas > 1 {
                    replicas = hpa.Spec.MinReplicas
                }
            }
        }
    } else if cd.Spec.AutoscalerRef.Kind == "ScaledObject" {
        so, err := c.flaggerClient.KedaV1alpha1().ScaledObjects(cd.Namespace).Get(context.TODO(), cd.Spec.AutoscalerRef.Name, metav1.GetOptions{})
        if err == nil {
            if so.Spec.MinReplicaCount != nil && *so.Spec.MinReplicaCount > 1 {
                replicas = so.Spec.MinReplicaCount
            }
        }
    }
```

I'm sorry it must've slipped past me in the previous review, but we need to check the `.spec.AutoscalerRef.Kind` before doing this. Since, we also support KEDA ScaledObjects now, we need to have the same logic for that as well.